File: Chapter02/python/packt2/spark/spark_submit_parser.py
from utilities01_py.helper_python import create_session
from utilities02_py.helper_python import extract_raw_records, parse_raw_warc
from globalp.python.packtg.helper_python_global import sample_warc_loc

if __name__ == "__main__":
    warc_loc = sample_warc_loc
    session = create_session(3, 'Submit Parser')

    raw_records = extract_raw_records(warc_loc, session)
    warc_records = raw_records \
        .flatMap(lambda record: parse_raw_warc(record))

    warc_records.toDF().printSchema()
    print(warc_records.count())

File: Chapter02/scala/packt2/spark/SparkSubmitParser.scala
package packt2.spark

import org.apache.hadoop.io.Text
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession
import Utilities02.WarcRecord
import Utilities01.HelperScala.createSession
import Utilities02.HelperScala.{sampleWarcLoc, extractRawRecords, parseRawWarc}

/**
 * Code for parsing .warc files of the WARC corpus
 *
 * @author Phil, https://github.com/g1thubhub
 */
object SparkSubmitParser {

  def main(args: Array[String]) = {

    val inputLocWarc = sampleWarcLoc
    implicit val session: SparkSession = createSession(3, "Submit Parser")

    val rawRecords: RDD[Text] = extractRawRecords(inputLocWarc)
    val warcRecords: RDD[WarcRecord] = rawRecords
      .flatMap(parseRawWarc(_))

    import org.apache.spark.sql._
    import session.implicits._
    val responses: DataFrame = warcRecords
      .filter(_.warcType == "response")
      .toDF()
    responses.printSchema()

    responses.show(3)

    val englishRecords = responses.filter($"language" === "en")
    println(englishRecords.count())

  }
}

